### PR TITLE
Api tests

### PR DESCRIPTION
**Pull request by AndriyPovzun**

### CODE REVIEW

File: API_Testing/gorest/api_collections/posts_collection.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import json
+from faker import Faker
+
+
+class PostsAPI(BaseAPI):
+    def __init__(self, env):
+        super().__init__(env)
+        self.__env = env
+        self.__posts_url = self.__env.post_url
+        self.__fake = Faker()
```
Comment (bohdandovbsyh) [inline API_Testing/gorest/api_collections/posts_collection.py]: try to initialize Faker in BaseAPI

File: API_Testing/gorest/api_collections/posts_collection.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                            "body": f'{self.__fake.text()}'}
+
+    def create_user_post(self, user_id, body=None):
+        self.__posts_url = self.__posts_url.replace('user_id', str(user_id))
+        if not body:
+            body = self.__fake_post
+        return self._post(url=self.__posts_url, body=body)
+
+    def get_all_user_posts(self, user_id):
+        self.create_user_post(user_id)
```
Comment (bohdandovbsyh) [inline API_Testing/gorest/api_collections/posts_collection.py:21]: why do you need to create this function?

File: API_Testing/gorest/conftest.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import pytest
+from API_Testing.gorest.api_collections.user_collection import User
+from API_Testing.gorest.data.user_data import UserData
+from API_Testing.gorest.CONSTANTS import ROOT_DIR
+from API_Testing.gorest.configurations.config_parser import Configuration
+import json
+
+
+@pytest.fixture()
```
Comment (bohdandovbsyh) [inline API_Testing/gorest/conftest.py]: it can has scope session

File: API_Testing/gorest/data/post_data.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        self.id = post_id
+        self.user_id = user_id
+        self.title = title
+        self.body = body
+
+    @classmethod
+    def from_json(cls, **kwargs):
+        return cls(post_id=kwargs['id'], user_id=kwargs['user_id'], title=kwargs['title'], body=kwargs['body'])
+
+    def is_valid_response(self):
```
Comment (bohdandovbsyh) [inline API_Testing/gorest/data/post_data.py:12]: it is not a good idea to make validation in this place

File: API_Testing/gorest/api_collections/posts_collection.py (hunk 1 above)
Comment (AndriyPovzun) [reply API_Testing/gorest/api_collections/posts_collection.py]: Done

File: API_Testing/gorest/conftest.py (hunk 3 above)
Comment (AndriyPovzun) [reply API_Testing/gorest/conftest.py]: done

File: API_Testing/gorest/api_collections/posts_collection.py (hunk 2 above)
Comment (AndriyPovzun) [reply API_Testing/gorest/api_collections/posts_collection.py:21]: The difference between get_all_user_posts and get_post_data_after_create is that the second method returns an instance of PostData. I can't check the response code, response reason, etc.

The first to analyze the response and other technical data. The second is for directly comparing validity or comparing fields. 

This is a useful method and there is a difference between the two

File: API_Testing/gorest/data/post_data.py (hunk 4 above)
Comment (AndriyPovzun) [reply API_Testing/gorest/data/post_data.py:12]: but where?

**Pull request merged**